src/Components/WebAssembly/JSInterop/src/WebAssemblyJSRuntime.cs

```diff
@@ -25,35 +25,20 @@ protected WebAssemblyJSRuntime()
     /// <inheritdoc />
     protected override string InvokeJS(string identifier, [StringSyntax(StringSyntaxAttribute.Json)] string? argsJson, JSCallResultType resultType, long targetInstanceId)
     {
-        var callInfo = new JSCallInfo
+        try
         {
-            FunctionIdentifier = identifier,
-            TargetInstanceId = targetInstanceId,
-            ResultType = resultType,
-            MarshalledCallArgsJson = argsJson ?? "[]",
-            MarshalledCallAsyncHandle = default
-        };
-
-        var result = InternalCalls.InvokeJS<object, object, object, string>(out var exception, ref callInfo, null, null, null);
-
-        return exception != null
-            ? throw new JSException(exception)
-            : result;
+            return InternalCalls.InvokeJSJson(identifier, targetInstanceId, (int)resultType, argsJson ?? "[]", 0);
+        }
+        catch (Exception ex)
+        {
+            throw new JSException(ex.Message, ex);
```

Comment:
I guess you are translating the exception for compatibility reasons ?

Reply:
Yep, `Microsoft.JSInterop.JSException` is how we previously exposed exceptions thrown from JavaScript.